index.php

```diff
@@ -32,108 +32,17 @@
 $download = optional_param('download', '', PARAM_RAW);
 $perpage = optional_param('perpage', 50, PARAM_INT);
 
+// Calls require_login and performs permissions checks for admin pages.
 admin_externalpage_setup('reportapocalypse', '', null, '',
     array('pagelayout' => 'report'));
 
-// This is an abitrary date based on the statements from browser developers relating to "mid 2019".
-$date = strtotime("2019-8-31 0:00");
+$title = get_string('pluginname', 'report_apocalypse');
+$url = new moodle_url("/report/apocalypse/index.php");
+$PAGE->set_url($url);
+$PAGE->set_context(context_system::instance());
+$PAGE->set_title($title);
+$PAGE->set_heading($title);
 
-// List of course category id's and their name to allow display in the report.
-$coursecategorynames = $DB->get_records_menu('course_categories', array(), '', 'id, name');
-
-$exportfilename = "flash-apocalypse-report";
-$table = new flexible_table('report_apocalypse');
-$table->show_download_buttons_at(array(TABLE_P_BOTTOM));
-
-if (!$table->is_downloading($download, $exportfilename)) {
-    $remaining = $date - time();
-    $daysremaining = floor($remaining / 86400);
-    echo $OUTPUT->header();
-    $imageurl = $CFG->wwwroot .'/report/apocalypse/pix/catalyst-logo.svg'; // Not using image_url for old site support.
-    echo '<span class="catalyst-logo">
-          <a href="https://www.catalyst.net.nz/products/moodle/?refer=report_apocalypse">
-          <img src="' . $imageurl . '" width="181"></a></span>';
-
-    if ($daysremaining > 0) {
-        echo $OUTPUT->heading(get_string('apocalypseinxdays', 'report_apocalypse', $daysremaining));
-    } else {
-        echo $OUTPUT->heading(get_string('apocalypseishere', 'report_apocalypse'));
-    }
-
-    echo $OUTPUT->box_start();
-    echo get_string('description', 'report_apocalypse');
-    echo $OUTPUT->box_end();
-}
-
-$table->define_baseurl($PAGE->url);
-$table->define_columns(array('category', 'coursefullname', 'component', 'name', 'html5'));
-$table->define_headers(array(
-    get_string('category'),
-    get_string("course"),
-    get_string('activitytype', 'report_apocalypse'),
-    get_string("activity"),
-    get_string('dualmode', 'report_apocalypse')
-));
-$table->sortable(true);
-$table->set_attribute('class', 'generaltable generalbox');
-$table->setup();
-
-// Check all modules in the site.
-$modules = $DB->get_records_menu('modules', array(), '', 'id, name');
-
-// Use table sort.
-$sort = $table->get_sql_sort();
-
-list($sql, $params) = report_apocalypse_sql($modules, $sort);
```

Comment:
I would probably have preferred to see this file stay mostly as-is and just replace this call to get the sql with a call to get the sql from the new cached table instead of getting it directly from the db.

Reply:
When the table is rendered, the call to `get_audit_results_paginated` only pulls out a select amount of records from the stored table of most recent results. 
The sql and parameters for the audit are only built when `audit_manager::run_audit()` is called, which is invoked by the scheduled task.
Is this what you meant?

Reply:
No - I would have preferred you left view.php mostly as-is, created a separate process to populate a new table, and then just swap out the sql here with a sql call on your new table - make more sense? - I can live with some of the changes you've made, but you've over-complicated a lot of code that could have been left very simple.

I really don't see the point of the apocalypse_datetime class and a lot of your "helper" functions just aren't needed. - get_activityurl/get_activityname/get_htmlpresent/delete_records etc etc. You should just be calling API functions directly, not putting simple API calls into a helper function and calling your helper function.

Reply:
Ok I've gone back and removed the function/methods which aren't needed and pulled the API calls out.
The `renderer` is where all of the components are built for display, and the `audit_table` class just extends the `table_sql` class to build the html table from our new database `report_apocalypse` table, it doesn't run the audit.
The actual audit conduct is managed by the `audit_manager` and a call to it from the scheduled task, `scheduled_flash_audit`.
Is there anything else you want me to refactor?